src/neat.js

```diff
@@ -184,278 +227,251 @@ Neat.prototype = {
+      }
     }
-  },
+  };
 
   /**
-   * Sets fitness scores for the current population members.
+   * Evaluates the current population, basically sets their `.score` property
+   *
+   * @return {Network} Fittest Network
    */
-  evaluate: async function () {
-    if(!this.dataset) {
-      throw new Error("You must include a dataset to use NEAT.evaluate()")
-    }
-    var i;
-    if (this.fitnessPopulation) {
-      if (this.clear) {
-        for (i = 0; i < this.population.length; i++) {
-          this.population[i].clear();
-        }
+  self.evaluate = async function (dataset) {
+    if (self.fitnessPopulation) {
+      if (self.clear) {
+        for (let i = 0; i < self.population.length; i++)
+          self.population[i].clear();
       }
-      await this.fitness(this.population, this.dataset);
+      await self.fitness(dataset, self.population);
     } else {
-      for (i = 0; i < this.population.length; i++) {
-        var genome = this.population[i];
-        if (this.clear) genome.clear();
-        genome.score = await this.fitness(genome, this.dataset);
+      for (let i = 0; i < self.population.length; i++) {
+        const genome = self.population[i];
+        if (self.clear) genome.clear();
+        genome.score = await self.fitness(dataset, genome);
+        self.population[i] = genome;
       }
     }
-  },
+    
+    // Sort the population in order of fitness
+    self.sort()
+
+    return self.population[0]
+  };
 
   /**
-   * Sorts the population by fitness
-   */
-  sort: function () {
-    this.population.sort(function (a, b) {
+   * Sorts the population by score
+  */
+  self.sort = function () {
+    self.population.sort(function (a, b) {
       return b.score - a.score;
     });
-  },
+  };
 
   /**
    * Returns the fittest genome of the current population
    *
    * @returns {Network} Current population's fittest genome
   */
-  getFittest: function () {
-    // Check if evaluated
-    if (typeof this.population[this.population.length - 1].score === 'undefined') {
-      this.evaluate();
-    }
-    if (this.population[0].score < this.population[1].score) {
-      this.sort();
-    }
+  self.getFittest = function () {
+    // Check if evaluated. self.evaluate is an async function
+    if (typeof self.population[self.population.length - 1].score === 'undefined')
+      self.evaluate();
+    
+    if (self.population[0].score < self.population[1].score) self.sort();
 
-    return this.population[0];
-  },
+    return self.population[0];
+  };
 
   /**
    * Returns the average fitness of the current population
    *
    * @returns {number} Average fitness of the current population
    */
-  getAverage: function () {
-    if (typeof this.population[this.population.length - 1].score === 'undefined') {
-      this.evaluate();
-    }
+  self.getAverage = function () {
+    if (typeof self.population[self.population.length - 1].score === 'undefined')
+      self.evaluate(); // self.evaluate is an async function
```

Comment:
Shouldn't this wait if this is an async method?

Reply:
Yep! I'm doing a second round of cleanup on Neat and this is top priority